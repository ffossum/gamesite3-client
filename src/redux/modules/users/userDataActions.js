/* @flow */
export const FETCHED_USER_DATA = 'data/fetched user(s)';

type FetchedUserDataAction = {
  type: typeof FETCHED_USER_DATA,
  payload: PublicUserData[],
};
export function fetchedUserData(
  users: PublicUserData[],
): FetchedUserDataAction {
  return {
    type: FETCHED_USER_DATA,
    payload: users,
  };
}

export type UserDataAction = FetchedUserDataAction;
