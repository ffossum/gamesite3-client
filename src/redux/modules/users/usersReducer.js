/* @flow */
import type { Action } from '../../actions';
import { FETCHED_USER_DATA } from './userDataActions';
import { LOGIN_SUCCESS } from '../login/loginActions';

export type UsersState = {
  [string]: PublicUserData,
};

const initialState = {};

export default function usersReducer(
  state: UsersState = initialState,
  action: Action,
) {
  switch (action.type) {
    case FETCHED_USER_DATA: {
      const newUsers = {};
      action.payload.forEach(user => {
        newUsers[user.id] = user;
      });

      return {
        ...state,
        ...newUsers,
      };
    }
    case LOGIN_SUCCESS: {
      const user = action.payload;
      return {
        ...state,
        [user.id]: user,
      };
    }
    default:
      return state;
  }
}
