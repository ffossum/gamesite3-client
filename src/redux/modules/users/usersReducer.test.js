/* @flow */
/* eslint-env jest */
import usersReducer from './usersReducer';
import { fetchedUserData } from './userDataActions';

describe('users reducer', () => {
  const initialState = usersReducer(undefined, { type: '@@INIT' });
  test('initial state', () => {
    expect(initialState).toMatchSnapshot();
  });
  test('fetched users are added to state', () => {
    const action = fetchedUserData([
      {
        id: 'asdf-id',
        username: 'asdf',
      },
      {
        id: 'zxcv-id',
        username: 'zxcv',
      },
    ]);
    const state = usersReducer(initialState, action);
    expect(state).toEqual({
      'asdf-id': {
        id: 'asdf-id',
        username: 'asdf',
      },
      'zxcv-id': {
        id: 'zxcv-id',
        username: 'zxcv',
      },
    });
  });
});
