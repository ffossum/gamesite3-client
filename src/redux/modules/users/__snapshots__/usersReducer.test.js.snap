// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`users reducer initial state 1`] = `Object {}`;
